src/App.tsx

```diff
@@ -1,204 +1,72 @@
-          </div>
-
-          {/* This todo is being edited */}
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Filter.ALL);
+  const [errorNotification, setErrorNotification] = useState<Errors>(
+    Errors.DEFAULT,
+  );
 
-            {/* This form is shown instead of the title and remove button */}
-            <form>
-              <input
-                data-cy="TodoTitleField"
-                type="text"
-                className="todo__title-field"
-                placeholder="Empty todo will be deleted"
-                value="Todo is being edited now"
-              />
-            </form>
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorNotification(Errors.LOADING);
+      });
+  }, []);
 
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+  useEffect(() => {
+    setTimeout(() => {
+      setErrorNotification(Errors.DEFAULT);
+    }, 3000);
+  });
 
-          {/* This todo is in loadind state */}
-          <div data-cy="Todo" className="todo">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-              />
-            </label>
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
 
-            <span data-cy="TodoTitle" className="todo__title">
-              Todo is being saved now
-            </span>
+  const getFilteredTodos = () => {
+    switch (filter) {
+      case Filter.ACTIVE:
+        return todos.filter(todo => !todo.completed);
 
-            <button type="button" className="todo__remove" data-cy="TodoDelete">
-              ×
-            </button>
+      case Filter.COMPLETED:
+        return todos.filter(todo => todo.completed);
 
-            {/* 'is-active' class puts this modal on top of the todo */}
-            <div data-cy="TodoLoader" className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
-        </section>
+      default:
+        return todos;
+    }
+  };
 
-        {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer" data-cy="Footer">
-          <span className="todo-count" data-cy="TodosCounter">
-            3 items left
-          </span>
+  const filteredTodos = getFilteredTodos();
```

Comment:
`useMemo`?
and you use this variable only in `TodoList`, so you can create it there